src/fs/filesystem.ts

```diff
@@ -201,14 +178,19 @@ export class FileSystem {
         resultPretty = await fn(this.prettyTree, relPath)
 
         this.publicTree = result
-        this.prettyTree = resultPretty as unknown as PublicTreeBare
+        this.prettyTree = resultPretty as unknown as BareTree
```

Comment:
This method keep getting bigger, but I'm not sure of a better way to do it. Will keep thinking :thinking:

Reply:
@expede, this is the method that I was talking about that gets more convoluted as we add more branches to the tree. I think we need some better abstraction here